sample-geovelocity/siddhi-execution-geovelocity/src/main/siddhi-files/IS_ANALYTICS_GEOVELOCITY.siddhi

```diff
@@ -0,0 +1,174 @@
+    latitude double,
+    longitude double,
+    city string,
+    country string,
+    authenticationSuccess bool);
+
+-- OverallAuthTableTable for all the authentication data.
+@store(type='rdbms', datasource='IS_ANALYTICS_DB', 'field.length=serviceProvider:100, identityProvider:100, userStoreDomain:50, region:45')
+@primaryKey('meta_tenantId', 'eventId', 'eventType')
+define table OverallAuthTable (
+    meta_tenantId int,
+    contextId string,
+    eventId string,
+    eventType string,
+    username string,
+    localUsername string,
+    userStoreDomain string,
+    tenantDomain string,
+    remoteIp string,
+    region string,
+    inboundAuthType string,
+    serviceProvider string,
+    rememberMeEnabled bool,
+    forceAuthEnabled bool,
+    passiveAuthEnabled bool,
+    rolesCommaSeparated string,
+    authenticationStep string,
+    identityProvider string,
+    authenticationSuccess bool,
+    authStepSuccess bool,
+    stepAuthenticator string,
+    isFirstLogin bool,
+    identityProviderType string,
+    utcTime string,
+    timestamp long
+);
+
+-- TravelRestrictedAreas database table to store restricted location combinations.
+@primaryKey('Id')
+@index('Id')
+@store(type='rdbms', datasource='IS_ANALYTICS_DB', 'field.length=username:20, loginTime:20, loginIp:20, city:20')
+define table TRAVELRESTRICTEDAREAS(
+    Id int,
+    currentLocation string,
+    lastLocation string);
+
+-- Function to calculate time gap between last login and current login attempt.
+define function timeGapFn[javascript] return int{
+    var timeGap = Math.floor(parseInt(data[0])-parseInt(data[1]));
+    return timeGap
+};
+
+-- Stream with input stream data.
+from  InputStream
+select messageId,
+    username,
+    loginTime,
+    loginIp
+insert into LoginDetailsStream;
+
+-- Stream to get all the login data by the IP address.
+from  LoginDetailsStream#geo:geocoordinate(loginIp)
+select messageId,
+    username,
+    loginTime,
+    loginIp,
+    latitude,
+    longitude,
+    geo:findCityFromIP(loginIp) as city,
+    geo:findCountryFromIP(loginIp) as country
+insert into FinalLoginDetailsStream;
+
+-- Update GEO_VELOCITY_INFO database table with the last successful login details.
+from FinalLoginDetailsStream#window.length(1) join OverallAuthTable
+on FinalLoginDetailsStream.username == OverallAuthTable.username and OverallAuthTable.authenticationSuccess == true
+select FinalLoginDetailsStream.username as username,
+    max(OverallAuthTable.timestamp) as loginTime,
+    true as authenticationSuccess
+update GEOVELOCITY_INFO
+    set GEOVELOCITY_INFO.loginTime=loginTime,GEOVELOCITY_INFO.authenticationSuccess=authenticationSuccess
+    on GEOVELOCITY_INFO.username == username;
+
+-- Get login details of last successful login and the current login attempt.
+from FinalLoginDetailsStream#window.length(1) join GEOVELOCITY_INFO
+on  FinalLoginDetailsStream.username==GEOVELOCITY_INFO.username and GEOVELOCITY_INFO.authenticationSuccess == true
```

Comment:
Shall we format all the comparisons in the siddhi file in a consistent way? At some places, there are spaces on either side and in some places, it's not.

FinalLoginDetailsStream.usernam**e==G**EOVELOCITY_INFO.username and GEOVELOCITY_INFO.authenticationSucces**s == t**rue

Reply:
Fixed in https://github.com/wso2/samples-is/pull/265